File: 2/ScalaProject/app/models/Product.scala
package models

import play.api.libs.json.Json

case class Product(id: Long, name: String, description: String, category: Int, price: Int)
object Product {
  implicit val productFormat = Json.format[Product]
}

case class NewProduct(name: String, description: String, category: Int, price: Int)
object NewProduct {
  implicit val productFormat = Json.format[NewProduct]
}

File: 4/models/product.go
package models

import (
	"gorm.io/gorm"
)

type Product struct {
	gorm.Model
	Name        string  `json:"name"`
	Description string  `json:"description"`
	CategoryID  uint64  `json:"category_id"`
	Price       float64 `json:"price"`
	Stock       uint64  `json:"stock"`
}
